WeakAuras/BuffTrigger2.lua

```diff
@@ -1129,8 +1129,10 @@ local function TriggerInfoApplies(triggerInfo, unit)
     return false
   end
 
-  if triggerInfo.raidRole and not triggerInfo.raidRole[WeakAuras.UnitRaidRole(controllingUnit) or ""] then
-    return false
+  if WeakAuras.IsClassicOrBCCOrWrath() then
+    if triggerInfo.raidRole and not triggerInfo.raidRole[WeakAuras.UnitRaidRole(controllingUnit) or ""] then
```

Comment:
I think it's better to make effectiveRaidRole nil on Retail in BuffTrigger.Add line 2595

Reply:
fixed in https://github.com/WeakAuras/WeakAuras2/pull/3785/commits/db9189b38104e0ccf6257456c8a5150f3d9367e5 where more fields are tested